python/cudf/cudf/core/column/categorical.py

```diff
@@ -916,19 +1026,21 @@ def sort_by_values(self, ascending=True, na_position="last"):
         )
         return col, inds
 
-    def element_indexing(self, index):
+    def element_indexing(self, index: int) -> ScalarObj:
         val = self.as_numerical.element_indexing(index)
-        return self._decode(val) if val is not None else val
+        return self._decode(int(val)) if val is not None else val
 
     @property
-    def __cuda_array_interface__(self):
+    def __cuda_array_interface__(self) -> Mapping[str, Any]:
         raise TypeError(
             "Categorical does not support `__cuda_array_interface__`."
             " Please consider using `.codes` or `.categories`"
             " if you need this functionality."
         )
 
-    def to_pandas(self, index=None, nullable=False):
+    def to_pandas(
+        self, index=None, nullable: bool = False, **kwargs
```

Comment:
Can we type `index`?

Reply:
There's a ton of places where things have been typed either `Any` or left untyped (equivalent to `Any`), where we should use `ColumnLike` instead. This is only one of them..

I've fixed to use `ColumnLike` here.